### PR TITLE
PATTERN_PROJECTS PATTERN_CONTEXT regex modified

### LINKED ISSUE #957: todotxt: Allow to insert tags/categories with multiple ++/@@
#### General information

* **App version:** 
* **System:** 

#### Description

I use some context / list entries starting with double @, like `@@WartenAuf`, `@@Irgendwann`, `@@Test`

The main reason is: these are some meta-lists, they should be ordered and marked separately and maybe there are some other reasons. 

This is conform to the todotxt syntax. But Markor has some issues with these entries. The list of list entries is strange and they can't be assigned from the list list to an entry if they are not right in the list. The list startes

- "
- Irgendwann
- Test
- WartenAuf
- abc"

Could you handle @@ (and maybe @@@, who knows)?

### PR DESCRIPTION
**Pull request by aisbergde**
fixes #957 

current definition:

    public static final Pattern PATTERN_PROJECTS = Pattern.compile("\\B(?:\\++)(\\S+)");
    public static final Pattern PATTERN_CONTEXTS = Pattern.compile("\\B(?:\\@+)(\\S+)");

I am not really sure what the `Patter.compile` does, but from my understanding there is some tranformation, maybe the final regex is `\B(?:\@+)(\S+)`

I testerd here: [Online regex tester and debugger: PHP, PCRE, Python, Golang and JavaScript](https://regex101.com/) or here [RegExr: Learn, Build, &amp; Test RegEx](https://regexr.com/)

Test cases:

```
@aaa
qwert @bbb
asd @@ccc
ghi @@@ddd
jkl a@bbb
mno c@@ddd
```

`\B(?:@+)(\S+)`

current definition, has issues

`\B\@(\@*\S+)`

issues with

> mno c@@ddd

`\s\@(\@*\S+)`

works best

### CODE REVIEW

**Comment by gsantner** [comment]
`c@@ddd` should not be marked, only when starts with @

**Comment by aisbergde** [comment]
You are right, it doesn't work when on the very first line :-(

I don't know how this all works in the program, can't compile and test. I tested only on these websites.
`\B` has some issues with double @@ inside a word

`\B\@(\@*\S+)`

>@aaa
>aaa@bbb
>ccc@@ddd

will be parsed as

>@aaa
>@ddd

but maybe this is OK in most cases?

This could do the job, but only if you parse per line, not if you parse a multi line text
`\s|^`

`\s|^\@(\@*\S+)`

But If I see how you parse begin of line in other patterns, this should work?:
    public static final Pattern PATTERN_PRIORITY_A = Pattern.compile("(?:^|\\n)\\(([Aa])\\)\\s");

Should I change or will you change wile merging?
Then this could be OK?

**Comment by aisbergde** [comment]
public static final Pattern PATTERN_PROJECTS = Pattern.compile("(?:^|\\n|\\s)\\+(\\+*\\S+)");
    public static final Pattern PATTERN_CONTEXTS = Pattern.compile("(?:^|\\n|\\s)\\@(\\@*\\S+)");

**Comment by aisbergde** [comment]
Sorry, I never really used all this regex staff, only for very simple expressions.

**Comment by harshad1** [comment]
This is the website i have been using to test my regexes https://www.freeformatter.com/java-regex-tester.html.

**Comment by gsantner** [comment]
I just temporary lock that one, to have discussion/comments at #957 (else it gets hard to follow)

**Comment by gsantner** [comment]
Closing as invalid.

As noted in issue, the regex is fine. Its only about inserting/loading into tag/project in dialog, where we strip +/@ and just add one.
